### PR TITLE
Added debug and source flags in sandbox

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added new flag `debug` to see logs of the sandbox or else it will stay quiet and show a loading bar
- Added new flag `flytesnacks-repo` to mount any repository (In flytesnacks example we need flytesnack code to be inside the sandbox container)
- Added cache in build workflow

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by evalsocket** [comment]
@pmahindrakar-oss Demo : https://asciinema.org/a/420583

File: cmd/sandbox/sandbox_util.go [hunk 1]
```diff
@@ -87,34 +90,44 @@ func startContainer(cli *client.Client) (string, error) {
 	if err != nil {
 		return "", err
 	}
-
-	if _, err := io.Copy(os.Stdout, r); err != nil {
-		return "", err
+	if sandboxConfig.DefaultConfig.Debug {
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:93]: pass the debug flag as parameter to the function and have the caller pass it from one place.

File: cmd/sandbox/sandbox_util.go [hunk 2]
```diff
@@ -87,34 +90,44 @@ func startContainer(cli *client.Client) (string, error) {
...
+			Target: "/etc/rancher/",
+		},
+		// TODO (Yuvraj) Add flytectl config in sandbox and mount with host file system
+		//{
+		//	Type:   mount.TypeBind,
+		//	Source: f.FilePathJoin(f.UserHomeDir(), ".flyte", "config.yaml"),
+		//	Target: "/.flyte/",
+		//},
+	}
+	if len(sandboxConfig.DefaultConfig.Source) > 0 {
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:112]: Same here  pass this as a param

File: cmd/sandbox/sandbox_util.go [hunk 3]
```diff
@@ -87,34 +90,44 @@ func startContainer(cli *client.Client) (string, error) {
...
+		volumes = append(volumes, mount.Mount{
+			Type:   mount.TypeBind,
+			Source: sandboxConfig.DefaultConfig.Source,
+			Target: "/usr/src",
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:116]: const . Also will this target change ever in the sandbox.

File: cmd/sandbox/sandbox_util.go [hunk 4]
```diff
@@ -147,14 +161,29 @@ func readLogs(cli *client.Client, id string) error {
 		return err
 	}
 	scanner := bufio.NewScanner(reader)
+	if !sandboxConfig.DefaultConfig.Debug {
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:164]: similarly here. and L174

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package sandbox
+
+//go:generate pflags Config --default-var DefaultConfig
+var (
+	DefaultConfig = &Config{
+		Debug: false,
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/sandbox/sandbox_config.go:6]: Lets keep the default as true , so that the user knows whats happening instead of being impatient and waiting.

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	}
+)
+
+// Config
+type Config struct {
+	Debug  bool   `json:"debug" pflag:", Enable debugging"`
+	Source string `json:"source" pflag:", Path of your repository"`
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/sandbox/sandbox_config.go:13]: Can we rename this to SnacksRepo

File: cmd/sandbox/sandbox_util.go (hunk 1 above)
Comment (evalsocket) [reply cmd/sandbox/sandbox_util.go:93]: done

**Review by pmahindrakar-oss — APPROVED** [review]
